lib/vmpooler/api.rb

```diff
@@ -16,37 +12,48 @@ def initialize
       JSON.pretty_generate(result)
     end
 
-    # Load dashboard components
-    begin
-      require 'dashboard'
-    rescue LoadError
-      require File.expand_path(File.join(File.dirname(__FILE__), 'dashboard'))
-    end
+    def self.execute(torun, config, redis, metrics)
+      self.settings.set :config, config
+      self.settings.set :redis, redis unless redis.nil?
+      self.settings.set :metrics, metrics
+      self.settings.set :checkoutlock, Mutex.new
 
-    use Vmpooler::Dashboard
+      if metrics.respond_to?(:setup_prometheus_metrics)
+        # Prometheus metrics are only setup if actually specified
+        # in the config file.
+        metrics.setup_prometheus_metrics
 
-    # Load API components
-    %w[helpers dashboard reroute v1].each do |lib|
-      begin
-        require "api/#{lib}"
-      rescue LoadError
-        require File.expand_path(File.join(File.dirname(__FILE__), 'api', lib))
+        use Rack::Deflater
```

Comment:
I think including this regardless of if Prometheus is enabled would be a good idea FWIW. See https://www.schneems.com/2017/11/08/80-smaller-rails-footprint-with-rack-deflate/ for context

Reply:
👍

Reply:
Done